docs/source/sparse.rst

```diff
@@ -1,146 +1,350 @@
+tensor of indices and a tensor of values. A sparse COO tensor can be
+constructed by providing these two tensors, as well as the size of the
+sparse tensor (which cannot be inferred from the indices and values
+tensors!). Suppose we want to define a sparse tensor with the entry 3
+at location (0, 2), entry 4 at location (1, 0), and entry 5 at
+location (1, 2).  We would then write:
+
+    >>> i = [[0, 1, 1],
+             [2, 0, 2]]
+    >>> v =  [3, 4, 5]
+    >>> s = torch.sparse_coo_tensor(i, v, (2, 3))
+    >>> s
+    tensor(indices=tensor([[0, 1, 1],
+                           [2, 0, 2]]),
+           values=tensor([3, 4, 5]),
+           size=(2, 3), nnz=3, layout=torch.sparse_coo)
+    >>> s.to_dense()
+    tensor([[0, 0, 3],
+            [4, 0, 5]])
 
-Note that the input to LongTensor is NOT a list of index tuples.  If you want
+Note that the input ``i`` is NOT a list of index tuples.  If you want
 to write your indices this way, you should transpose before passing them to
 the sparse constructor:
 
-    >>> i = torch.LongTensor([[0, 2], [1, 0], [1, 2]])
-    >>> v = torch.FloatTensor([3,      4,      5    ])
-    >>> torch.sparse.FloatTensor(i.t(), v, torch.Size([2,3])).to_dense()
-     0  0  3
-     4  0  5
-    [torch.FloatTensor of size 2x3]
-
-You can also construct hybrid sparse tensors, where only the first n
-dimensions are sparse, and the rest of the dimensions are dense.
-
-    >>> i = torch.LongTensor([[2, 4]])
-    >>> v = torch.FloatTensor([[1, 3], [5, 7]])
-    >>> torch.sparse.FloatTensor(i, v).to_dense()
-     0  0
-     0  0
-     1  3
-     0  0
-     5  7
-    [torch.FloatTensor of size 5x2]
+    >>> i = [[0, 2], [1, 0], [1, 2]]
+    >>> v =  [3,      4,      5    ]
+    >>> s = torch.sparse_coo_tensor(list(zip(*i)), v, (2, 3))         # or
+    >>> s = torch.sparse_coo_tensor(torch.tensor(i).t(), v, (2, 3))
+    >>> torch.sparse_coo_tensor(i.t(), v, torch.Size([2,3])).to_dense()
+    tensor([[0, 0, 3],
+            [4, 0, 5]])
+
+Sparse COO tensors support indexing operations (except slicing in
+sparse dimensions):
+
+    >>> s[1]
+    tensor(indices=tensor([[0, 2]]),
+           values=tensor(  [4, 5]),
+           size=(3,), nnz=2, layout=torch.sparse_coo)
+    >>> s[1, 0]
+    tensor(4)
 
 An empty sparse tensor can be constructed by specifying its size:
 
-    >>> torch.sparse.FloatTensor(2, 3)
-    SparseFloatTensor of size 2x3 with indices:
-    [torch.LongTensor with no dimension]
-    and values:
-    [torch.FloatTensor with no dimension]
+    >>> torch.sparse_coo_tensor(size=(2, 3))
+    tensor(indices=tensor([], size=(2, 0)),
+           values=tensor([], size=(0,)),
+           size=(2, 3), nnz=0, layout=torch.sparse_coo)
+
+or by specifying sparse layout to :func:`torch.empty`:
+
+    >>> torch.empty(2, 3, layout=torch.sparse_coo)
+    tensor(indices=tensor([], size=(2, 0)),
+           values=tensor([], size=(0,)),
+           size=(2, 3), nnz=0, layout=torch.sparse_coo)
```

Comment:
I would remove this as it is unintuitive, or replace it by `torch.zeros`.

Reply:
This section is about constructing an empty sparse tensor. In the context of torch (as well as numpy), an "empty tensor" is a tensor with allocated but uninitialized data storage. In the case of sparse tensor, the pre-allocation of data storage does not make sense. So, for sparse tensors, `torch.empty` is equivalent to `torch.zeros`, that is, empty sparse tensor does not specify any elements and with the default zero fill value, the empty sparse tensor represents a tensor with all zero elements.
That said when a fill value attribute will be introduced, the plan is to keep the empty-zeros equivalence, see https://github.com/pytorch/rfcs/pull/8#discussion_r495487183 .

I'll remove this as suggested.